docs/proposals/JNG-XXXX-by-class-runtime-caching.md

```
@@ -0,0 +1,360 @@
+| Test instance (for injection)  | ❌ — recreated per test by JUnit |
+
+Therefore the following derived artifacts are safe to cache once per class:
+
+| Artifact                       | Cacheable | Notes |
+|--------------------------------|-----------|-------|
+| `QueryFactory`                 | ✅        | pure function of the model |
+| First Liquibase migration      | ✅        | run once, subsequent are no-ops |
+| `databaseModule`               | ✅        | pure function of dialect + datasource |
+| Guice `Injector`               | ✅        | depends on the above + class-level `@JudoTest` annotation |
+| `PlatformTransactionManager`   | ✅        | resolved from injector once |
+| Interceptor instances          | ⚠️ depends on stateful behaviour — see §6 |
```

markdownlint-cli2 (0.22.1)

[warning] Table column count
Expected: 3; Actual: 2; Too few cells, row will be missing data

(MD056, table-column-count)